src/main/java/cs2103/v15_1j/jimjim/ui/SearchPaneController.java

```diff
@@ -109,16 +110,15 @@ private void showSearchResults(){
 			searchGridPane.addColumn(3, dateTimeLabel);
 		}
 
-		counter = 0;
-		for(FloatingTask task: lists.getFloatingTasksList()){
-			counter++;
+		for(FloatingTask task: searchResultsList.getFloatingTasksList()){
 			CheckBox cb = new CheckBox();
 			cb.setDisable(true);
 			cb.selectedProperty().bindBidirectional(task.completedProperty());
 			searchGridPane.addColumn(0, cb);
 			cb.setOnMouseClicked(event -> showSearchResults());
 
-			Label idLabel = new Label("[F"+counter+"]");
+			int id = masterList.getFloatingTasksList().indexOf(task) + 1;
```

Comment:
no need. just use `masterList.indexOf` it will automatically search the right list based on the class